src/odemis/gui/cont/multi_point_correlation.py

```diff
@@ -0,0 +1,815 @@
+        fm_coords = fm_coords[:min_n]
+        poi_coord = self.correlation_target.fm_pois
+        poi_coord_2d = self.correlation_target.fm_streams[0].getPixelCoordinates_alt(poi_coord.coordinates.value[0:2])
+        poi_coords.append([poi_coord_2d[0], poi_coord_2d[1], poi_coord.coordinates.value[2]])
+        poi_coords = numpy.array(poi_coords, dtype=numpy.float32)
+        # Run the correlation
+        self.correlation_target.correlation_result = run_tdct_correlation(fib_coords=fib_coords, fm_coords=fm_coords,
+                                                                          poi_coords=poi_coords,
+                                                                          fib_image=fib_da, fm_image=fm_image,
+                                                                          path=path)
+        # Update the output pameters of the correlation
+        self._tab_data_model.projected_points = []
+        self.correlation_target.fib_projected_fiducials = []
+        points = self.correlation_target.correlation_result['output']['error']['reprojected_3d']
+        for n, i in enumerate(points[0]):
+            p_pos = self.correlation_target.fib_stream.getPhysicalCoordinates((points[0][n], points[1][n]))
+            target = Target(x=p_pos[0], y=p_pos[1], z=0, name="PP" + str(n + 1), type="ProjectedPoints", index=n + 1,
+                            fm_focus_position=0)
+            self._tab_data_model.projected_points.append(target)
+            self.correlation_target.fib_projected_fiducials.append(target)
+
+        projected_poi = get_reprojected_poi_coordinate(self.correlation_target.correlation_result)
+        projected_poi_target = Target(x=projected_poi[0], y=projected_poi[1], z=0, name="PPOI", type="ProjectedPOI",
+                                      index=1,
+                                      fm_focus_position=0)
+        self._tab_data_model.projected_points.append(projected_poi_target)
+        self.correlation_target.fib_projected_pois = [projected_poi_target]
+        # Display the output in the relevant views
+        for vp in self._viewports:
+            if vp.view.name.value == "SEM Overview":
+                vp.canvas.update_drawing()
+
+    def _on_char(self, evt: wx.Event) -> None:
+        """
+        Handle key presses on the views
+        :param evt: (wx.Event) the event
+        """
+        # TODO change the keys to Ctrl + Left Click and Alt + Left Click
+        key = evt.GetKeyCode()
+        ctrl_mode = evt.ControlDown()
+
+        # pass through event, if not a valid correlation key or enabled
+        valid_keys = [wx.WXK_LEFT, wx.WXK_RIGHT, wx.WXK_UP, wx.WXK_DOWN]
+        if key not in valid_keys:
+            evt.Skip()
+            return
+
+        if ctrl_mode:
+            if key == wx.WXK_UP:
+                if self._tab_data_model.focussedView.value.name.value == "FLM Overview" or self._tab_data_model.focussedView.value.name.value == "SEM Overview":
```

Comment:
Don't rely on view name to identify the views. Either use the `stream_classes` of the view, or compare to the `viewport.view`.

Anyway, for this specific code, as there are only 2 views, probably  you don't even need a `if`?